hadoop-hdds/config/src/main/java/org/apache/hadoop/hdds/conf/ConfigurationReflectionUtil.java

```diff
@@ -240,4 +242,49 @@ private static ConfigType detectConfigType(Class<?> parameterType,
       }
     }
   }
+
+  public static Optional<String> getDefaultValue(Class<?> configClass,
+      String fieldName) {
+    return findFieldConfigAnnotationByName(configClass, fieldName)
+        .map(Config::defaultValue);
+  }
+
+  public static Optional<String> getKey(Class<?> configClass,
+      String fieldName) {
+    ConfigGroup configGroup =
+        configClass.getAnnotation(ConfigGroup.class);
+
+    return findFieldConfigAnnotationByName(configClass,
+        fieldName).map(
+            config -> configGroup == null ? config.key()
+                : configGroup.prefix() + "." + config.key());
+  }
+
+  public static Optional<ConfigType> getType(Class<?> configClass,
+      String fieldName) {
+    return findFieldConfigAnnotationByName(configClass, fieldName)
+        .map(Config::type);
+  }
+
+  private static Optional<Config> findFieldConfigAnnotationByName(
+      final Class<?> configClass, String fieldName) {
+    Class<?> theClass = configClass;
+    while (theClass != null) {
+      Optional<Config> config = Stream.of(theClass.getDeclaredFields())
+          .filter(f -> f.getName().equals(fieldName))
+          .findFirst()
+          .map(f -> f.getAnnotation(Config.class));
+
+      if (config.isPresent()) {
+        return config;
+      }
+
+      if (!theClass.getSuperclass().equals(Object.class)) {
+        theClass = theClass.getSuperclass();
+      } else {
+        theClass = null;
+      }
```

Comment:
The following change makes this method safe to invoke with `Object.class` directly (current code produces NPE).  It may be unlikely that someone does this, so I think it's OK to commit as is.

```suggestion
      theClass = theClass.getSuperclass();
      if (Object.class.equals(theClass)) {
        theClass = null;
      }
```